src/test/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateStepTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Test
+    public void wrongDiskPoolId() throws Exception {
+        setupDiskPool();
+        createUpstreamJobAndRun(DISK_POOL_ID + "random");
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disk Pool ID matching '%s' was found in the global config", DISK_POOL_ID + "random"), upstreamRun);
+    }
+
+    @Test
+    public void emptyDisks() throws Exception {
+        setupDiskPool();
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disks were defined in the global config for Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingDiskId() throws Exception {
+        setupDiskPool(new Disk("", "name", "mount", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Disk ID was not provided in the Jenkins global config for the Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingMasterMountPoint() throws Exception {
+        setupDiskPool(new Disk(DISK_ID, "name", "", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Mounting point from Master to the disk is not defined for Disk ID '%s', from Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingPhysicalPathOnDisk() throws Exception {
+        setupDiskPool(new Disk(DISK_ID, "name", "mount", ""));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Physical path on disk was not provided in the Jenkins global config for Disk ID: '%s', within Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void physicalPathOnDiskNotRelative() throws Exception {
+        setupDiskPool(new Disk(DISK_ID, "name", "mount", "/path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Physical path on disk defined for Disk ID '%s', within Disk Pool ID '%s' must be a relative path", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void successfullyAllocateWorkspace() throws Exception {
+        Disk disk = new Disk(DISK_ID, "name", "mount", "path");
+        setupDiskPool(disk);
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatusSuccess(upstreamRun);
+        j.assertLogContains(format("Selected Disk ID '%s' from the Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+        j.assertLogContains(format("The path on Disk is: %s/%s/%d", disk.getPhysicalPathOnDisk(), upstreamRun.getParent().getFullDisplayName(), upstreamRun.getNumber()), upstreamRun);
+    }
+
+    /* ##### Tests for the downstream Job ###### */
```

Comment:
This is something I'll take into consideration when multiple features will be implemented for the `exwsAllocate` step.